react/features/base/dialog/components/MessageDialog.web.js

```diff
@@ -0,0 +1,170 @@
+import React, { Component } from 'react';
+
+import { Label } from '@atlaskit/field-base';
+import Toggle from '@atlaskit/toggle';
+
+import { DIALOG_PROP_TYPES } from '../constants';
+import Dialog from './Dialog';
+
+import { translate } from '../../i18n';
+import jitsiLocalStorage from '../../../../../modules/util/JitsiLocalStorage';
+
+/**
+ * Implements a message dialog with an option do not show again component.
+ */
+class MessageDialog extends Component {
+    /**
+     * {@code MessageDialog} component's property types.
+     *
+     * @static
+     */
+    static propTypes = {
+        ...DIALOG_PROP_TYPES,
+
+        /**
+         * This is the body of the dialog, the component children. (optional)
+         */
+        children: React.PropTypes.node,
+
+        /**
+         * A (optional) message to be displayed along with children components
+         * of the dialog. This is the key to be stored in local storage
+         * to indicate that this dialog should not be shown again if desired by
+         * the user.
+         */
+        doNotShowWarningLocalStorageKey: React.PropTypes.string,
+
+        /**
+         * A (optional) message to be displayed along with children components
+         * of the dialog.
+         */
+        doNotShowWarningTextKey: React.PropTypes.string,
+
+        /**
+         * A (optional) message to be displayed along with children components
+         * of the dialog.
+         */
+        message: React.PropTypes.string,
+
+        /**
+         * Invoked to obtain translated strings.
+         */
+        t: React.PropTypes.func
+    };
+
+    /**
+     * Initializes a new {@code MessageDialog} instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        // Bind event handlers so they are only bound once for every instance.
+        this._onDontShowChanged = this._onDontShowChanged.bind(this);
+        this._onSubmit = this._onSubmit.bind(this);
+    }
+
+    /**
+     * Load from local storage the option does user had indicated that we should
+     * not show this message dialog.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentWillMount() {
+        if (this.props.doNotShowWarningLocalStorageKey) {
+            this.setState({
+                doNotShowDialog:
+                    jitsiLocalStorage.getItem(
+                        this.props.doNotShowWarningLocalStorageKey)
+                            === 'true'
+            });
+        } else {
+            this.setState({
+                doNotShowDialog: false
+            });
+        }
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        if (this.state.doNotShowDialog) {
+            return null;
+        }
```

Comment:
Returning null here would leave the app in a state where a reference to the modal component is left in the redux store. Do you see that causing any problems at this stage of modal development?